lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -360,48 +381,27 @@ private VerifiableCredentialBundle getCredentialBundle(String userId, String evc
                 evcsService.getVerifiableCredentialsByState(
                         userId, evcsAccessToken, CURRENT, PENDING_RETURN);
 
-        if (vcs.isEmpty()) {
-            return new VerifiableCredentialBundle(List.of(), false);
-        }
+        var isPendingReturn = !isNullOrEmpty(vcs.get(PENDING_RETURN));
+
+        // + inherited VCs
+        var evcsIdentityVcs =
+                new ArrayList<>(
+                        vcs.getOrDefault(CURRENT, List.of()).stream()
+                                .filter(vc -> HMRC_MIGRATION.equals(vc.getCri()))
+                                .toList());
 
-        // Use pending return vcs to determine identity if available
-        var evcsIdentityVcs = vcs.get(PENDING_RETURN);
-        var isPending = true;
-        if (isNullOrEmpty(evcsIdentityVcs)) {
-            evcsIdentityVcs = vcs.get(CURRENT);
-            isPending = false;
+        if (isPendingReturn) {
+            // + vcs deferred from making an identity by an async CRI
+            evcsIdentityVcs.addAll(vcs.getOrDefault(PENDING_RETURN, List.of()));
```

Comment:
Might be able to tidy this logic a little by just adding the inherited VCs here, otherwise return all? (if I'm following it correctly!)